static/js/subs.js

```diff
@@ -657,6 +657,15 @@ export function setup_page(callback) {
             throttled_redraw_left_panel();
         });
 
+        $("#stream_filter input[type='text']").on("keypress", (e) => {
+            // Disallow opening of stream-creation modal which restricts the hash
+            // property to update, for users who don't have permission to create streams.
+            if (e.which === 13 && !settings_data.user_can_create_streams()) {
+                e.preventDefault();
+                return;
+            }
+        });
```

Comment:
I think this is probably the wrong fix -- ideally we'd find whatever Enter handler is trying to create a stream on that keypress, and change its behavior, rather than adding a new handler to override it.

Reply:
Probably this handler code:
```
    $("#subscriptions_table").on("keyup", ".subscriber_list_add form", (e) => {                 
        if (e.which === 13) {                                                                   
            e.preventDefault();                                                                 
            submit_add_subscriber_form(e);                                                      
        }                                                                                       
    });                                                                                         
```

Reply:
> Probably this handler code:
> 
> ```
>     $("#subscriptions_table").on("keyup", ".subscriber_list_add form", (e) => {                 
>         if (e.which === 13) {                                                                   
>             e.preventDefault();                                                                 
>             submit_add_subscriber_form(e);                                                      
>         }                                                                                       
>     });                                                                                         
> ```

If I am not mistaken, isn't this handler responsible for adding new members within an existing stream ? ( Since the `.subscriber_list_add` class lies within the `subscriber_list_settings` container)

Reply:
> ideally we'd find whatever Enter handler is trying to create a stream on that keypress

Yup, But there isn't an handler which is specifically targeted towards the `Enter` key for the filter stream input bar. 
In fact, the <a href = 'https://github.com/zulip/zulip/blob/935a7b18cb2720288370d75aefcf1d0146631913/static/js/subs.js#L983'>`click`</a> event is responsible for also accounting the `Enter` hotkey (Manually checked it).

Hence, to bind the `Enter` hotkey with Filter stream input bar, the following methodology can also be implemented:

```javascript
$("#stream_filter input[type='text']").on("keydown", (e) => {
  // Debounce filtering in case a user is typing quickly
  if (e.which === 13 && !settings_data.user_can_create_streams()) {
    e.preventDefault();
    return;
  }
  throttled_redraw_left_panel();
});
```

ie, to manually change the <a href = 'https://github.com/zulip/zulip/blob/935a7b18cb2720288370d75aefcf1d0146631913/static/js/subs.js#L655'>`input`</a> handle of filter stream input bar.

Thoughts @timabbott -- Whether to proceed with the above code or the one that is already implemented in the pr ?

Reply:
I still think we're missing something here -- hitting "Enter" on a random webpage doesn't reload the browser, so we must have some code that is causing this to happen.

Reply:
Cool, Will try to debug it deeply!